File: 07-03-2026/perfect_cube_count.js
function countPerfectCubes(a, b) {
  let [maxNum, minNum, absMax, count] = [
    Math.max(a, b),
    Math.min(a, b),
    Math.max(Math.abs(a), Math.abs(b)),
    0,
  ];
  let [negFalse, posFalse] = [true, true];

  for (let i = 0; i <= absMax; i++) {
    if (!negFalse && !posFalse) break;

    if (minNum < 0 && negFalse) {
      const negCube = -(i ** 3);
      if (negCube < 0 && negCube >= minNum) count++;
      if (negCube < minNum) negFalse = false;
    }

    if (maxNum > 1 && posFalse) {
      const cube = i ** 3;
      if (cube <= maxNum && cube >= minNum) count++;
      if (cube > maxNum) posFalse = false;
    }
  }

  return count;
}

countPerfectCubes(3, 30); // 2
countPerfectCubes(1, 30); // 3
countPerfectCubes(30, 0); // 4
countPerfectCubes(-64, 64); // 9
countPerfectCubes(9214, -8127); // 41

File: 07-03-2026/perfect_cube_count.py
def count_perfect_cubes(a, b):
    maxNum, minNum, absMax, count = max(a, b), min(a, b), max(abs(a), abs(b)), 0
    negFalse, posFalse = True, True

    for i in range(absMax):
        if not negFalse and not posFalse:
            break

        if minNum < 0 and negFalse:
            negCube = -(i ** 3);
            if negCube < 0 and negCube >= minNum: 
                count+=1;
            if negCube < minNum: 
                negFalse = False;

        if maxNum > 1 and posFalse:
            cube = i ** 3;
            if cube <= maxNum and cube >= minNum:
                count+=1;
            if cube > maxNum:
                posFalse = False;
    
    return count


print(count_perfect_cubes(3, 30)); # 2
print(count_perfect_cubes(1, 30)); # 3
print(count_perfect_cubes(30, 0)); # 4
print(count_perfect_cubes(-64, 64)); # 9
print(count_perfect_cubes(9214, -8127)); # 41
